### PR TITLE
Moved all '!xkcd' command-related functions to xkcd.py and formatted '!xkcd' help menu

### PR DESCRIPTION
**Pull request by tofu-tigre**

### CODE REVIEW

**Comment by robertquitt** [comment]
FYI--you dont need to close the old PR and make a new one, pushing more commits will update it.

**Review by aninrusimha — COMMENTED** [review]
Overall looks good, have you tested this on a local server?

**Comment by tofu-tigre** [comment]
@aninrusimha Yeah, I tested it on a local server. Seemed to work perfectly!

**Comment by aninrusimha** [comment]
@tofu-tigre Awesome, I'm going to leave this up for now as an example of a pull request for Root Day, but I'll merge after that! Lets meet up some time to look at the github issues.

**Review by aninrusimha — APPROVED** [review]

File: apps/discordbot/bot.py [hunk 1]
```diff
@@ -91,14 +87,12 @@ async def on_message(self, message):
...
-                await message.channel.send("Title: " + str(rand_comic["title"]))
-                await message.channel.send("Issue: " + str(rand_comic["issue"]))
-                await message.channel.send("Description: *" + str(rand_comic["image_text"]) + "*")
-                await message.channel.send(str(rand_comic["image_url"]))
+            # Validate "!xkcd" command
+            if xkcd.is_valid_xkcd_command(msg):
+                await xkcd.get_xkcd(message)
             else:
-                await message.channel.send("Sorry! I couldn't find a comic right now. Try again later.")
+                await message.channel.send("Please ensure that your command is properly formatted. Type `!xkcd -help` for more information.")
```
Comment (robertquitt) [inline apps/discordbot/bot.py:94]:
```suggestion
                await message.channel.send("Please ensure that your command is properly formatted. Type `!xkcd --help` for more information.")

```

File: apps/discordbot/xkcd.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+
+def is_valid_xkcd_command(msg):
+    """ Returns whether msg contains an valid request. """
+
+    arguments = msg.split()
+    if len(arguments) > MAX_ARGUMENT_LENGTH or len(arguments) < MIN_ARGUMENT_LENGTH:
+        return False
+    elif arguments[1] in VALID_XKCD_COMMANDS:
+        if arguments[1] == "-issue" or arguments[1] == "-i":
```
Comment (robertquitt) [inline apps/discordbot/xkcd.py:18]:
```suggestion
        if arguments[1] == "--issue" or arguments[1] == "-i":
```

File: apps/discordbot/xkcd.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+
+def get_current():
+    return get_json(HOST)
+
+
+def get_json(url):
+    try:
+        return requests.get(url + "info.0.json").json()
+    except json.JSONDecodeError:
```
Comment (robertquitt) [inline apps/discordbot/xkcd.py:83]:
```suggestion
    except json.decoder.JSONDecodeError:
```

File: apps/discordbot/xkcd.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+async def get_xkcd(message):
+    """
+    Displays either a random XKCD comic, a specific issue, or the '!xkcd' help panel.
+    Assumes that a valid command is contained within msg (checked by is_valid_xkcd_command(String)).
+    """
+    msg = message.content.lower().split()
+    cmd = msg[1]
+    comic = None
+    
+    if cmd == "-help" or cmd == "-h":
```
Comment (robertquitt) [inline apps/discordbot/xkcd.py:35]:
```suggestion
    if cmd == "--help" or cmd == "-h":
```

**Pull request merged**